plugins/janus_streaming.c

```diff
@@ -4000,7 +4000,7 @@ static json_t *janus_streaming_process_synchronous_request(janus_streaming_sessi
 			}
 			if(data) {
 				const char *datafile = json_string_value(data);
-				drc = janus_recorder_create(NULL, "text", (char *)datafile);
+				drc = janus_recorder_create(NULL, source->textdata ? "text" : "bin", (char *)datafile);
```

Comment:
This is wrong. The recorder currently only supports "text", and will fail if you pass "bin" (or "binary"). Since we'll have to fix this in the core, please modify the PR so that only "text" is supported for the moment.

Reply:
Corrected.